### PR TITLE
Remove service on termination when exec 'kubectl run' command with flags "--rm" and "--expose"

### LINKED ISSUE #40504: kubectl run with --rm and --expose don't remove service on termination
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Kubernetes version** (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"08e099554f3c31f6e6f07b448ab3ed78d0520507", GitTreeState:"clean", BuildDate:"2017-01-12T07:31:07Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"08e099554f3c31f6e6f07b448ab3ed78d0520507", GitTreeState:"clean", BuildDate:"2017-01-12T04:52:34Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}

**Environment**:
- **Cloud provider or hardware configuration**: Self-Hosted
- **OS** (e.g. from /etc/os-release): Ubuntu
- **Kernel** (e.g. `uname -a`): Linux 4.4.0-59-generic #80-Ubuntu SMP Fri Jan 6 17:47:47 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
- **Install tools**: kubeadm
- **Others**:

**What happened**:
I'm create pod with command: `kubectl run go-http-test -it --rm --port 5000 --expose=true --image=tanapolsh/go-http-test --restart=Never`, it also create service for an option `--expose=true` but once I terminate pod, service don't be removed along with pod which cause next execution of the same command would get this error `Error from server (AlreadyExists): services "go-http-test" already exists`

**What you expected to happen**:
kubectl should remove the service on termination.

**How to reproduce it** (as minimally and precisely as possible):
* create pod using `kubectl run go-http-test -it --rm --port 5000 --expose=true --image=tanapolsh/go-http-test --restart=Never` and wait for pod to started
* Press `CTRL + C` and wait for pod is terminated
* run command `kubectl run go-http-test -it --rm --port 5000 --expose=true --image=tanapolsh/go-http-test --restart=Never` again, so you will get error

**Anything else do we need to know**:

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:
As the title says and issue #40504 mentioned.
cc @tanapoln

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #40504 

**Special notes for your reviewer**:
Related to: #44915

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/run.go [hunk 1]
```diff
@@ -88,6 +88,14 @@ var (
 		kubectl run pi --schedule="0/5 * * * ?" --image=perl --restart=OnFailure -- perl -Mbignum=bpi -wle 'print bpi(2000)'`))
 )
 
+type RunObject struct {
+	Object  runtime.Object
+	Mapper  meta.RESTMapper
+	Mapping *meta.RESTMapping
+}
+
+var runObjectMap = map[string]*RunObject{}
```
Comment (soltysh) [inline pkg/kubectl/cmd/run.go]: Do not use global vars, if you need to pass this between methods either do it directly, or create a struct methods and update the internal state of it, similarly to how it's done in other commands.

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/kubectl/cmd/run.go (hunk 1 above)
Comment (xiangpengzhao) [inline pkg/kubectl/cmd/run.go]: Thanks @soltysh ! I also didn't think global vars is a good solution. Will address.

**Comment by xiangpengzhao** [comment]
@soltysh inline comments addressed. PTAL. Thanks!

**Comment by spiffxp** [comment]
@k8s-bot ok to test

**Comment by xiangpengzhao** [comment]
@k8s-bot bazel test this

**Review by soltysh — APPROVED** [review]

**Comment by soltysh** [comment]
Thanks!
/lgtm

**Comment by xiangpengzhao** [comment]
ping @adohe for approve :)

**Comment by xiangpengzhao** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this
@k8s-bot kops aws e2e test this

**Comment by xiangpengzhao** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by xiangpengzhao** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by xiangpengzhao** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this

**Comment by xiangpengzhao** [comment]
rebased. @soltysh PTAL
and
ping @adohe for approval
Thanks!

**Comment by xiangpengzhao** [comment]
ping @soltysh and @adohe for lgtm and approval. Thanks!

**Comment by xiangpengzhao** [comment]
Since didn't receive any response from @adohe 
/unassign @adohe 
and
/assign @deads2k 
for approval

Thanks!

**Comment by xiangpengzhao** [comment]
Thanks @deads2k for approval!
Seems like bot only recognizes a `/lgtm` without any strings following it, so please give another `/lgtm` :)

**Comment by xiangpengzhao** [comment]
@deads2k Wow, thanks for so quick response!

**Pull request merged**